distributedlog-core/src/main/java/com/twitter/distributedlog/BKSyncLogReaderDLSN.java

```diff
@@ -116,7 +110,7 @@ public void resumeReadAhead() {
 
     @Override
     public void onSuccess(LogRecordWithDLSN record) {
-        setLastSeenDLSN(record.getDlsn());
+        this.lastSeenDLSN = record.getDlsn();
```

Comment:
lastSeenDLSN is referenced below in a multi statement block - are you certain atomicity isn't required there?